modules/gdscript/gdscript_parser.cpp

```diff
@@ -7571,6 +7571,10 @@ GDScriptParser::DataType GDScriptParser::_reduce_identifier_type(const DataType
 	}
 
 	if (_get_member_type(base_type, p_identifier, member_type)) {
+		if (current_function && current_function->_static) {
+			_set_error("Can't access non-static variable (\"" + p_identifier.operator String() + "\") from a static function.", p_line);
```

Comment:
The "non-static" may imply GDScript has static variables, which isn't the case. I'm not sure if the proposed wording is better, but it may be a tad less confusing:

```suggestion
			_set_error("Can't access member variable (\"" + p_identifier.operator String() + "\") from a static function.", p_line);
```

Reply:
I agree with @Calinou , "non-static" implies that static variables exist somewhere.